controlplane/kubeadm/controllers/controller_test.go

```diff
@@ -384,7 +384,7 @@ func TestReconcileClusterNoEndpoints(t *testing.T) {
 	result, err := r.Reconcile(ctrl.Request{NamespacedName: util.ObjectKey(kcp)})
 	g.Expect(err).NotTo(HaveOccurred())
 	// this first requeue is to add finalizer
-	g.Expect(result).To(Equal(ctrl.Result{Requeue: true}))
+	g.Expect(result).To(Equal(ctrl.Result{Requeue: false, RequeueAfter: 20 * time.Second}))
```

Comment:
If we don't add the `RequeueAfter` when we are only patching for the finalizer, I would expect that we should be able to match against `ctrl.Result{}` here